### DISCUSSION TITLE
manage-users realm management role is a way too broad!

### DISCUSSION THREAD

**Discussion by mindarchitect** [keycloak/keycloak#14454] [Q&A]
### Description

Hi
Keycloak 19.0.1

Maybe I am missing something but can not find a way to restrict newly created client role to user creation only. Seems like manage-user do the job but from security perspective it is way too broad.

How can i reduce role scope only to standard CRUD actions for service account roles?
I use only Keycloak REST API + Spring Boot Adapter for authorization.

P.S.

Keycloak has very poor practical examples docs. It is really hard to understand those concepts without meaningful app examples..
I do not understand why this decision was made to combine all possible user management actions into 1 role only. What if I do not need to read users, delete them etc? Why it was done in such a "all or nothing" way???

Why not introduce standard CRUD roles instead?

Regards,
Andrey

**Comment by Alexcei88** [comment]
I also want to have a specific role for creation of user. In current moment for creation of user I need to provide manage-users role. This role is very broad

**Reply by Alexcei88** [reply]
I think it is duplicate discussion. There is a similiar [discussion](https://github.com/keycloak/keycloak/discussions/13604)
